app/views/posts/show.html.erb

```diff
@@ -1,52 +1,43 @@
-<div class="card my-5">
-  <div class="card-body">
-    <p id="notice"><%= notice %></p>
-
-    <h1 class="text-center"><%= @post.title %></h1>
-    <hr >
-    <div class="my-4">
-      <%= simple_format(@post.body) %>
+<div class="container mt-4">
+  <h1><%= @post.title %></h1>
+  <p class="text-muted">
+    Posted by <%= @post.user.email %> on <%= @post.created_at.strftime("%B %d, %Y") %>
+  </p>
+  <p><%= @post.body %></p>
+  
+      <h2>Comments</h2>
+  
+      <% @comments.each do |comment| %>
+       <div class="card mb-3">
+          <div class="card-body">
+          <p><%= comment.body %> - <small class="text-muted">Posted by <%= comment.user.email %></small></p>
+            <div class="button-group">
+              <%= link_to 'Edit Comment', edit_comment_path(comment), class: "btn btn-warning btn-sm" %> |
+              <%= link_to 'Reply', new_comment_path(post_id: @post.id, parent_id: comment.id), class: "btn btn-info btn-sm" %> |
+              <%= button_to 'Delete Comment', comment_path(comment), method: :delete, data: { confirm: 'Are you sure?' }, class: "btn btn-danger btn-sm" %>
+            </div>
+          </div>
+          <div class="card-footer">
+            <% comment.replies.each do |reply| %>
```

Comment:
Since your replies are being stored as comments, it's causing the replies to show up twice on the page. Once underneath the comment that it is replying to, and once as it's own comment to the post. My suggestion for preventing this would be to either make a separate model for the Reply and have that belong to the comment and user who made the reply, or add an attribute to the comments that indicates whether it's a reply or not. With the additional attribute solution, you could then set up a conditional statement in your view to only show a comment if it isn't marked as a reply, and then show the replies only once on the comment it is replying to!